src/main/java/org/sasanlabs/service/vulnerability/ssrf/SSRFVulnerability.java

```diff
@@ -1,52 +1,221 @@
 package org.sasanlabs.service.vulnerability.ssrf;
 
-import com.nimbusds.jose.util.StandardCharset;
-import java.io.InputStream;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
 import java.net.URL;
 import java.net.URLConnection;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.regex.Pattern;
 import org.sasanlabs.internal.utility.LevelConstants;
+import org.sasanlabs.internal.utility.Variant;
 import org.sasanlabs.internal.utility.annotations.AttackVector;
 import org.sasanlabs.internal.utility.annotations.VulnerableAppRequestMapping;
 import org.sasanlabs.internal.utility.annotations.VulnerableAppRestController;
 import org.sasanlabs.service.vulnerability.bean.GenericVulnerabilityResponseBean;
 import org.sasanlabs.vulnerability.types.VulnerabilityType;
+import org.springframework.core.io.ClassPathResource;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.util.StreamUtils;
 import org.springframework.web.bind.annotation.RequestParam;
 
 @VulnerableAppRestController(descriptionLabel = "SSRF_VULNERABILITY", value = "SSRFVulnerability")
 public class SSRFVulnerability {
 
-    private static final String IMAGE_URL = "imageurl";
-    private static final transient Logger LOGGER = LogManager.getLogger(SSRFVulnerability.class);
+    private static final String FILE_URL = "fileurl";
+    private static final String FILE_PATH = "static/files/SSRF.txt";
```

Comment:
There is no need to make the challenge here as our target audience is scanner. I would still suggest to have just one button in UI or even no button in UI and as we load the level, we do an Api call to backend which will load the name of projects by SasanLabs.
Every level will send the url from where to load the projects from user input except the last one which is super secure.

Reply:
I would still suggest to have just one button in UI or even no button in UI and as we load the level, we do an Api call to backend which will load the name of projects by SasanLabs.
Every level will send the url from where to load the projects from user input except the last one which is super secure.

Reply:
so from your side nothing will change in backend except in front end you will be sending a fixed url always.